prov/efa/test/efa_unit_test_use_device_rdma.c

```diff
@@ -0,0 +1,118 @@
+#include "efa_unit_tests.h"
+
+
+void test_use_device_rdma(
+	const int env_val,
+	const int setopt_val,
+	const int expected_val,
+	const uint32_t api_version)
+{
+	int ret = 0;
+	struct fi_info *hints, *info;
+	struct fid_fabric *fabric = NULL;
+	struct fid_domain *domain = NULL;
+	struct fid_ep *ep = NULL;
+	struct rxr_ep *rxr_ep;
+	bool rdma_capable_hw;
+	char env_str[8];
+
+	if (env_val >= 0) {
+		snprintf(env_str,7,"%d",env_val);
+		setenv("FI_EFA_USE_DEVICE_RDMA",env_str, 1);
+	} else {
+		unsetenv("FI_EFA_USE_DEVICE_RDMA");
+	}
+
+	hints = efa_unit_test_alloc_hints(FI_EP_RDM);
```

Comment:
Any reason we cannot use `efa_unit_test_resource_construct`? I

Reply:
hm, we do need to do getinfo with a couple different versions.  maybe we add an argument to resource_construct?

Reply:
> maybe we add an argument to resource_construct?

I agree - or provide/overload another entry point and keep the current callers unchanged.

Reply:
oh shoot.  We also skip() the test early on instances without hardware RDMA.  Not sure how to do that cleanly.  Any other ideas how to check for hardware support?

Reply:
If that is known at compile time we use a macro and skipped versions of the tests https://github.com/ofiwg/libfabric/blob/main/prov/efa/test/efa_unit_test_cq.c#L283

If it is inferred at runtime then we use the skip function from cmocka https://github.com/ofiwg/libfabric/blob/main/prov/efa/test/efa_unit_test_device.c#L14

Reply:
I am using the skip() function, but halfway through the resource creation.  If I let the endpoint get created with the wrong environment value the code will abort() rather than skip().

I don't see an easy way to re-use that function.